### PR TITLE
MCLOUD-6732: Fix issue with app/etc/env.php on build phase and remove named volumes

### PR DESCRIPTION
**Pull request by shiftedreality**
### Description

Fixed 2 issues:

- Having app/etc/env.php will result in the broken build phase since services are not available
- Replaced named volumes with regular, to avoid issues on Linux and WSL2

### Fixed Issues (if relevant)

1. https://jira.corp.magento.com/browse/MCLOUD-6732
2. #275

### Manual testing scenarios

Build PHP images locally

1. Test in Developer mode
2. Test in Production mode

### Release notes

- Fixed issue when app/etc/env.php was breaking build process if present.
- Replaced named volume with regular volumes to avoid issues on Linux and WSL2

### Associated documentation updates

Add link to Magento DevDocs PR or Issue, if needed.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] Pull request introduces user-facing changes and includes meaningful release notes and documentation
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

**Review by aepod — DISMISSED** [review]
Reviewed, this looks good. I did not check the outputted composer.json, but the changes all look perfect. This is what is required to make it work with WSL2

File: src/Compose/DeveloperBuilder.php [hunk 1]
```diff
@@ -93,44 +85,26 @@ public function build(Config $config): Manager
...
         ];
 
+        if (in_array($syncEngine, [self::SYNC_ENGINE_MUTAGEN, self::SYNC_ENGINE_DOCKER_SYNC], true)) {
+            if ($syncEngine === self::SYNC_ENGINE_DOCKER_SYNC) {
+                $syncConfig = ['external' => true];
+            } elseif ($syncEngine === self::SYNC_ENGINE_MUTAGEN) {
+                $syncConfig = [];
+            }
+
+            $volumesList[$volumePrefix . self::VOLUME_MAGENTO_SYNC] = $syncConfig;
```
Comment (oshmyheliuk) [inline src/Compose/DeveloperBuilder.php]: This code can be simplify
```php
$volumesList[$volumePrefix . self::VOLUME_MAGENTO_SYNC] = $syncEngine === self::SYNC_ENGINE_DOCKER_SYNC ? ['external' => true] : []
```

Comment (aepod) [reply src/Compose/DeveloperBuilder.php]: No it cannot, it is not a if/else it is if/elseif. The tertiary you have would create a problem. This volume should not exist if sync engine is not set to Dockersync or Mutagen.

**Comment by damienwebdev** [comment]
LGTM as well. Given that this PR appears to cover multiple behaviors, it may be worth splitting the diffs into appropriately addressed chunks and rebase merging or merging with merge commit instead of a squash.

Fixes  #214

File: src/Compose/DeveloperBuilder.php (hunk 1 above)
Comment (shiftedreality) [reply src/Compose/DeveloperBuilder.php]: This is fine, there is `if`-statement right above to check if this volume is needed

Comment (oshmyheliuk) [reply src/Compose/DeveloperBuilder.php]: But there are only two possible variants for `$syncEngine`
```
in_array($syncEngine, [self::SYNC_ENGINE_MUTAGEN, self::SYNC_ENGINE_DOCKER_SYNC], true)
```
so if/elseif has no sense after the previous `if` condition.

File: src/Compose/BuilderInterface.php [hunk 2]
```diff
@@ -39,7 +39,9 @@ interface BuilderInterface
     public const NETWORK_MAGENTO = 'magento';
     public const NETWORK_MAGENTO_BUILD = 'magento-build';
 
-    public const VOLUME_MAGENTO = 'magento';
+    public const VOLUME_MAGENTO = '.';
+    public const VOLUME_DOCKER_MNT = '.docker/mnt';
+    public const VOLUME_MARIADB_CONF = '.docker/mysql/mariadb.conf.d';
     public const VOLUME_MAGENTO_VENDOR = 'magento-vendor';
     public const VOLUME_MAGENTO_GENERATED = 'magento-generated';
     public const VOLUME_MAGENTO_VAR = 'magento-var';
```
Comment (arhiopterecs) [inline src/Compose/BuilderInterface.php]: It looks like these constants are no longer used:
- VOLUME_MAGENTO_VAR
- VOLUME_MAGENTO_ETC
- VOLUME_MAGENTO_STATIC
- VOLUME_MAGENTO_MEDIA

File: src/Compose/DeveloperBuilder.php [hunk 3]
```diff
@@ -92,45 +84,22 @@ public function build(Config $config): Manager
...
         $volumesList = [
-            $volumePrefix . self::VOLUME_MAGENTO_SYNC => $syncConfig,
             $volumePrefix . self::VOLUME_MAGENTO_DB => []
         ];
 
+        if (in_array($syncEngine, [self::SYNC_ENGINE_MUTAGEN, self::SYNC_ENGINE_DOCKER_SYNC], true)) {
+            $volumesList[$volumePrefix . self::VOLUME_MAGENTO_SYNC] = $syncEngine === self::SYNC_ENGINE_DOCKER_SYNC
+                ? ['external' => true]
+                : [];
+        }
```
Comment (arhiopterecs) [inline src/Compose/DeveloperBuilder.php]:
```suggestion
        $volumes = [self::VOLUME_MAGENTO . ':' . self::DIR_MAGENTO . ':delegated'];

        if (in_array($syncEngine, [self::SYNC_ENGINE_MUTAGEN, self::SYNC_ENGINE_DOCKER_SYNC], true)) {
            $volumesList[$volumePrefix . self::VOLUME_MAGENTO_SYNC] = $syncEngine === self::SYNC_ENGINE_DOCKER_SYNC
                ? ['external' => true]
                : [];
            $volumes = [$volumePrefix . self::VOLUME_MAGENTO_SYNC . ':' . self::DIR_MAGENTO . ':nocopy'];
        }
```

And remove the method `getMagentoVolumes()`

File: src/Compose/BuilderInterface.php (hunk 2 above)
Comment (shiftedreality) [reply src/Compose/BuilderInterface.php]: Removed

**Review by oshmyheliuk — APPROVED** [review]

**Review by arhiopterecs — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
QA approved

**Pull request merged**